test/jdk/javax/swing/JComboBox/bug4530952.java

```diff
@@ -0,0 +1,144 @@
+ * @key headful
+ * @run main bug4530952
+ */
+
+public class bug4530952 {
+    static JFrame frame;
+    static JButton btnAction;
+    static volatile Point loc;
+
+    private static boolean flag = false;
+
+    private static boolean passed() {
+        return flag;
+    }
+
+    private static void pass() {
+        flag = true;
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            Robot robot = new Robot();
+            SwingUtilities.invokeAndWait(() -> createTestUI());
+            robot.waitForIdle();
+            robot.delay(250);
+
+            // enter some text in combo box
+            robot.keyPress(KeyEvent.VK_A);
+            robot.keyRelease(KeyEvent.VK_A);
+            robot.keyPress(KeyEvent.VK_A);
+            robot.keyRelease(KeyEvent.VK_A);
+            robot.waitForIdle();
+            robot.delay(250);
+
+            // find and click action button
+            SwingUtilities.invokeAndWait(() -> loc = btnAction.getLocationOnScreen());
+            robot.waitForIdle();
+            robot.delay(250);
+
+            robot.mouseMove(loc.x, loc.y);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.waitForIdle();
+            robot.delay(250);
+
+            if (!passed()) {
+                throw new RuntimeException("Failed: button action was not fired");
+            }
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    public static void createTestUI() {
+        frame = new JFrame("bug4530952");
+        frame.setLayout(new FlowLayout());
+
+        btnAction = new JButton("Action");
+        JComboBox cmbAction = new JComboBox();
+
+        ActionListener al = e -> pass();
+        DocumentListener dl = new DocumentListener() {
+            @Override
+            public void changedUpdate(DocumentEvent evt) {
+                resetButtons();
+            }
+
+            @Override
+            public void insertUpdate(DocumentEvent evt) {
+                resetButtons();
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent evt) {
+                resetButtons();
+            }
+        };
+
+        // Add an editable combo box
+        cmbAction.setEditable(true);
+        frame.add(cmbAction);
+
+        btnAction.setEnabled(false);
+        frame.add(btnAction);
+
+        btnAction.addActionListener(al);
+        ((JTextField) cmbAction.getEditor().getEditorComponent()).
+                getDocument().addDocumentListener(dl);
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    public static void resetButtons() {
+        int length = ((JTextField) cmbAction.getEditor().getEditorComponent()).
```

Comment:
Test fails since cmbAction is declared in testUI ..it needs to be a static variable.